identity/src/commonMain/kotlin/com/android/identity/util/ByteStringUtil.kt

```diff
@@ -0,0 +1,277 @@
+    append((value shr 16).toByte())
+    append((value shr 24).toByte())
+}
+
+fun ByteStringBuilder.appendInt64(value: Long) {
+    append((value shr 56).toByte())
+    append((value shr 48).toByte())
+    append((value shr 40).toByte())
+    append((value shr 32).toByte())
+    append((value shr 24).toByte())
+    append((value shr 16).toByte())
+    append((value shr 8).toByte())
+    append((value shr 0).toByte())
+}
+
+fun ByteStringBuilder.appendInt64Le(value: Long) {
+    append((value shr 0).toByte())
+    append((value shr 8).toByte())
+    append((value shr 16).toByte())
+    append((value shr 24).toByte())
+    append((value shr 32).toByte())
+    append((value shr 40).toByte())
+    append((value shr 48).toByte())
+    append((value shr 56).toByte())
+}
+
+fun ByteStringBuilder.appendUInt8(value: Int) {
+    require(value in UByte.MIN_VALUE.toInt()..UByte.MAX_VALUE.toInt()) { "Value $value is out of UInt8 range" }
+    append(value.toByte())
+}
+
+fun ByteStringBuilder.appendUInt16(value: Int) {
+    require(value in UShort.MIN_VALUE.toInt()..UShort.MAX_VALUE.toInt()) { "Value $value is out of UInt16 range" }
+    append((value shr 8).toByte())
+    append(value.toByte())
+}
+
+fun ByteStringBuilder.appendUInt16Le(value: Int) {
+    require(value in UShort.MIN_VALUE.toInt()..UShort.MAX_VALUE.toInt()) { "Value $value is out of UInt16 range" }
+    append(value.toByte())
+    append((value shr 8).toByte())
+}
+
+fun ByteStringBuilder.appendUInt32(value: Int) {
+    require(value >= 0) { "Value $value is out of UInt32 range" }
+    append((value shr 24).toByte())
+    append((value shr 16).toByte())
+    append((value shr 8).toByte())
+    append((value shr 0).toByte())
+}
+
+fun ByteStringBuilder.appendUInt32Le(value: Int) {
+    require(value >= 0) { "Value $value is out of UInt32 range" }
+    append((value shr 0).toByte())
+    append((value shr 8).toByte())
+    append((value shr 16).toByte())
+    append((value shr 24).toByte())
+}
+
+fun ByteStringBuilder.appendUInt64Be(value: Long) {
+    require(value >= 0) { "Value $value is out of UInt64 range" }
+    append((value shr 56).toByte())
+    append((value shr 48).toByte())
+    append((value shr 40).toByte())
+    append((value shr 32).toByte())
+    append((value shr 24).toByte())
+    append((value shr 16).toByte())
+    append((value shr 8).toByte())
+    append((value shr 0).toByte())
+}
+
+fun ByteStringBuilder.appendUInt64Le(value: Long) {
+    require(value >= 0) { "Value $value is out of UInt64 range" }
+    append((value shr 0).toByte())
+    append((value shr 8).toByte())
+    append((value shr 16).toByte())
+    append((value shr 24).toByte())
+    append((value shr 32).toByte())
+    append((value shr 40).toByte())
+    append((value shr 48).toByte())
+    append((value shr 56).toByte())
+}
+
+// Readers
+
+fun ByteString.getInt8(offset: Int): Byte {
+    require(offset in 0 until size) { "Offset $offset is out of bounds for ByteString of size $size" }
```

Comment:
For clarity (here and below, with the matching type), maybe:
require(offset in 0..(size - Byte.SIZE_BYTES)) { ... }

...probably also prefixed with:
require(size >= Byte.SIZE_BYTES) { ... }